### PR TITLE
Remove support for reading 1.9 tx logs

### PR DESCRIPTION
**Pull request by davidegrohmann**
Since we don't support migration from 1.9 to 3.0, there is no need to
keep around code to read old 1.9 tx logs.

### CODE REVIEW

File: community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/log/entry/VersionAwareLogEntryReader.java [hunk 1]
```diff
@@ -45,18 +44,11 @@
 {
     // Exists for backwards compatibility until we drop support for one of the two versions (1.9 and 2.0)
```
Comment (lutovich) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/transaction/log/entry/VersionAwareLogEntryReader.java]: This comment should probably be removed as well.

File: community/kernel/src/main/java/org/neo4j/storageengine/api/CommandReaderFactory.java [hunk 2]
```diff
@@ -30,9 +30,7 @@
      * introduce its own scheme.
      *
      * @param version log entry version. Versions are typically 0 or negative numbers.
-     * @param legacyHeaderVersion legacy log header version for breaking tie between multiple versions
-     * of value {@code 0}. Will go away when 1.9 format becomes unsupported.
      * @return {@link CommandReader} for reading commands of that version.
      */
-    CommandReader byVersion( byte version, byte legacyHeaderVersion );
+    CommandReader byVersion( byte version );
```
Comment (lutovich) [inline community/kernel/src/main/java/org/neo4j/storageengine/api/CommandReaderFactory.java:35]: There exists only single implementation of `CommandReaderFactory` interface - `RecordStorageCommandReaderFactory`. Maybe we can collapse them?

File: community/kernel/src/main/java/org/neo4j/kernel/impl/recovery/RecoveryRequiredChecker.java [hunk 3]
```diff
@@ -66,8 +65,7 @@ public boolean isRecoveryRequiredAt( File dataDir ) throws IOException
         PhysicalLogFiles logFiles = new PhysicalLogFiles( dataDir, fs );
 
         LogEntryReader<ReadableClosablePositionAwareChannel> reader =
-                new VersionAwareLogEntryReader<>( LogEntryVersion.CURRENT.byteCode(),
-                        new RecordStorageCommandReaderFactory() );
+                new VersionAwareLogEntryReader<>( new RecordStorageCommandReaderFactory() );
```
Comment (lutovich) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/recovery/RecoveryRequiredChecker.java]: `VersionAwareLogEntryReader` is always constructed in the same way. Maybe add a default 0-arg constructor with creates or uses singleton `RecordStorageCommandReaderFactory` inside?

**Comment by lutovich** [comment]
@davidegrohmann looks good, made only couple minor comments.

File: community/kernel/src/main/java/org/neo4j/storageengine/api/CommandReaderFactory.java (hunk 2 above)
Comment (davidegrohmann) [inline community/kernel/src/main/java/org/neo4j/storageengine/api/CommandReaderFactory.java:35]: Not sure, the interface is part of the storageengine api, the other one is an internal implementation. Should we really make the internal class part of the api?

Comment (lutovich) [inline community/kernel/src/main/java/org/neo4j/storageengine/api/CommandReaderFactory.java:35]: No, probably not than.

**Pull request merged**